cmd/contour/servecontext_test.go

```diff
@@ -363,288 +364,165 @@ func TestParseHTTPVersions(t *testing.T) {
+					Address:   "0.0.0.0",
+					Port:      8080,
+					AccessLog: "/dev/stdout",
+				},
+				HTTPSListener: &contour_api_v1alpha1.EnvoyListener{
+					Address:   "0.0.0.0",
+					Port:      8443,
+					AccessLog: "/dev/stdout",
 				},
-				EnableExternalNameService: false,
-				RateLimitService:          nil,
-				Policy: &contour_api_v1alpha1.PolicyConfig{
-					RequestHeadersPolicy:  &contour_api_v1alpha1.HeadersPolicy{},
-					ResponseHeadersPolicy: &contour_api_v1alpha1.HeadersPolicy{},
-					ApplyToIngress:        false,
+				Health: &contour_api_v1alpha1.HealthConfig{
+					Address: "0.0.0.0",
+					Port:    8002,
 				},
-				Metrics: contour_api_v1alpha1.MetricsConfig{
+				Metrics: &contour_api_v1alpha1.MetricsConfig{
 					Address: "0.0.0.0",
-					Port:    8000,
+					Port:    8002,
+				},
+				ClientCertificate: nil,
+				Logging: &contour_api_v1alpha1.EnvoyLogging{
+					AccessLogFormat:       contour_api_v1alpha1.EnvoyAccessLog,
+					AccessLogFormatString: "",
+					AccessLogLevel:        contour_api_v1alpha1.LogLevelInfo,
+					AccessLogFields: contour_api_v1alpha1.AccessLogFields([]string{
+						"@timestamp",
+						"authority",
+						"bytes_received",
+						"bytes_sent",
+						"downstream_local_address",
+						"downstream_remote_address",
+						"duration",
+						"method",
+						"path",
+						"protocol",
+						"request_id",
+						"requested_server_name",
+						"response_code",
+						"response_flags",
+						"uber_trace_id",
+						"upstream_cluster",
+						"upstream_host",
+						"upstream_local_address",
+						"upstream_service_time",
+						"user_agent",
+						"x_forwarded_for",
+					}),
+				},
+				DefaultHTTPVersions: nil,
+				Timeouts: &contour_api_v1alpha1.TimeoutParameters{
+					ConnectionIdleTimeout: pointer.StringPtr("60s"),
+					ConnectTimeout:        pointer.StringPtr("2s"),
+				},
+				Cluster: &contour_api_v1alpha1.ClusterParameters{
+					DNSLookupFamily: contour_api_v1alpha1.AutoClusterDNSFamily,
+				},
+				Network: &contour_api_v1alpha1.NetworkParameters{
+					EnvoyAdminPort:    pointer.Int(9001),
+					XffNumTrustedHops: contourconfig.UInt32Ptr(0),
 				},
 			},
+			Gateway: nil,
+			HTTPProxy: &contour_api_v1alpha1.HTTPProxyConfig{
+				DisablePermitInsecure: pointer.Bool(false),
+				FallbackCertificate:   nil,
+			},
+			EnableExternalNameService: pointer.Bool(false),
+			RateLimitService:          nil,
+			Policy: &contour_api_v1alpha1.PolicyConfig{
+				RequestHeadersPolicy:  &contour_api_v1alpha1.HeadersPolicy{},
+				ResponseHeadersPolicy: &contour_api_v1alpha1.HeadersPolicy{},
+				ApplyToIngress:        pointer.Bool(false),
+			},
+			Metrics: &contour_api_v1alpha1.MetricsConfig{
+				Address: "0.0.0.0",
+				Port:    8000,
+			},
+		}
+	}
+
+	cases := map[string]struct {
+		getServeContext         func(ctx *serveContext) *serveContext
+		getContourConfiguration func(cfg contour_api_v1alpha1.ContourConfigurationSpec) contour_api_v1alpha1.ContourConfigurationSpec
+	}{
```

Comment:
I switched this test to use the pattern of "have a base config, then each test adds its own modification to that config" which I think works really well here for clarity & maintaintability and saves us ~600 lines of copy-pasted config.

Reply:
yep, good stuff